src/julia_internal.h

```diff
@@ -497,8 +497,20 @@ STATIC_INLINE jl_value_t *jl_gc_alloc_(jl_ptls_t ptls, size_t sz, void *ty)
     const size_t allocsz = sz + sizeof(jl_taggedvalue_t);
     if (sz <= GC_MAX_SZCLASS) {
         int pool_id = jl_gc_szclass(allocsz);
-        jl_gc_pool_t *p = &ptls->heap.norm_pools[pool_id];
         int osize = jl_gc_sizeclasses[pool_id];
+        jl_gc_pool_t *p = &ptls->heap.norm_pools[pool_id];
```

Comment:
What's the performance impact of this? I would expect it to be noticable.

Reply:
Yes, this should go in the constructor for CodeInstance and MethodInstance etc. instead of being a branch, if we end up adopting this patch.